src/main/java/org/datadog/jmxfetch/reporter/Reporter.java

```diff
@@ -157,11 +156,9 @@ public void sendMetrics(List<Metric> metrics, String instanceName, boolean canon
     }
 
     /** Submits service check. */
-    public void sendServiceCheck(String checkName, String status, String message, String[] tags) {
+    public void sendServiceCheck(String checkName, String serviceCheckName,
+                                 String status, String message, String[] tags) {
```

Comment:
I think we can/should remove `checkName` from the signature:

```suggestion
    public void sendServiceCheck(String serviceCheckName, String status,
                                 String message, String[] tags) {
```

(`incrementServiceCheckCount` seems to be only used to increment an internal counter only used in tests, let's make it track the full service check names instead of simply the check names, and update the tests accordingly)

Reply:
The service check count is also used for reporting here: https://github.com/DataDog/jmxfetch/blob/a4a17db0a778ee5cdccf290b892db8a81812b655/src/main/java/org/datadog/jmxfetch/App.java#L808 

It seems the intent of the reported service check count is about `the total of service checks sent for a specific check name` (not only about `X.can_connect` service check). So, if we have more service checks sent later, it seems to make sense to count by `check name` as it's currently done.

Reply:
oh, good catch, hadn't seen its use in `App.java`.

That said, it looks like the service check count is then set on the stats of the instance, not the check as a whole (see https://github.com/DataDog/jmxfetch/blob/37fc55f929f897066014e89946e666f03ee27eff/src/main/java/org/datadog/jmxfetch/Status.java#L104), so it's better to actually report the number of service checks sent for that specific instance of the check

Reply:
Seems there is a bug here, looks like the `service_check_count` is always reporting 0:

````

========
JMXFetch
========

  Initialized checks
  ==================
    confluent_platform
      instance_name : confluent_platform-localhost-31000
      message : <no value>
      metric_count : 26
      service_check_count : 0
      status : OK
      instance_name : confluent_platform-localhost-31001
      message : <no value>
      metric_count : 312
      service_check_count : 0
      status : OK
      instance_name : confluent_platform-localhost-31002
      message : <no value>
      metric_count : 205
      service_check_count : 0
      status : OK
      instance_name : confluent_platform-localhost-31008
      message : <no value>
      metric_count : 205
      service_check_count : 0
      status : OK
      instance_name : confluent_platform-localhost-31006
      message : <no value>
      metric_count : 29
      service_check_count : 0
      status : OK
      instance_name : confluent_platform-localhost-31007
      message : <no value>
      metric_count : 91
      service_check_count : 0
      status : OK
      instance_name : confluent_platform-localhost-31004
      message : Unable to instantiate or initialize instance localhost:31004. Is the target JMX Server or JVM running? java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.ServiceUnavailableException [Root exception is java.rmi.ConnectException: Connection refused to host: localhost; nested exception is:
	java.net.ConnectException: Connection refused (Connection refused)]
      metric_count : 0
      service_check_count : 0
      status : ERROR
```

> That said, it looks like the service check count is then set on the stats of the instance, not the check as a whole 

Agree this that. Do you mind if we adresse that in a separate PR (and the potential bug mentioned above)? It seems out of the scope of this PR.

Reply:
hmm, thanks for noticing this. Let's address this separately then agreed, thanks for adding a card with context in our backlog